src/core/main.c

```diff
@@ -1253,37 +1253,39 @@ static void bump_file_max_and_nr_open(void) {
 #endif
 
 #if BUMP_PROC_SYS_FS_NR_OPEN
-        int v = INT_MAX;
-
-        /* Argh! The kernel enforces maximum and minimum values on the fs.nr_open, but we don't really know
-         * what they are. The expression by which the maximum is determined is dependent on the architecture,
-         * and is something we don't really want to copy to userspace, as it is dependent on implementation
-         * details of the kernel. Since the kernel doesn't expose the maximum value to us, we can only try
-         * and hope. Hence, let's start with INT_MAX, and then keep halving the value until we find one that
-         * works. Ugly? Yes, absolutely, but kernel APIs are kernel APIs, so what do can we do... 🤯 */
+        /* cf. https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/file.c?h=v6.8#n27
+	 * for the progeny of the below value for `sysctl_nr_open_max`. Note
+	 * that the below logic was first introduced in `git` commit
+	 * eceea0b3df05ed262ae32e0c6340cc7a3626632d of the `linux` kernel and
+	 * was designed to prevent overflows when determining the maximum
+	 * number of possible file descriptors. */
+	#define BITS_PER_LONG __WORDSIZE
+        #define __const_min(x, y) ((x) < (y) ? (x) : (y))
+        unsigned sysctl_nr_open_max =
+            __const_min(INT_MAX, ~(size_t)0/sizeof(void *)) & -BITS_PER_LONG;
+
+        unsigned v = sysctl_nr_open_max;
```

Comment:
This is completely redundant.

Reply:
I don't understand. I don't normally write C/C++. How should I initialize the value of `v`?

Reply:
Tried to address this in feae976644074aa8cdb3a5564534407ea63076ed but I'm not exactly sure what you want @yuwata .